Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -1181,18 +1185,185 @@ async def create_incidents(incidents: list, user_name: str, user_email: str, use
+    :param channel_id: str: The ID of the channel for the message event we recieved.
+    :param text: str: The text of the message event to be sent to a war room.
+    :param user: AsyncSlackResponse: The user object for the user who sent the message.
+    :return: None
+    """
+    integration_context = fetch_context()
+    demisto.info("Fetching context")
+    if not integration_context or 'mirrors' not in integration_context:
+        return
+
+    mirrors = json.loads(integration_context['mirrors'])
+    mirror_filter = list(filter(lambda m: m['channel_id'] == channel_id, mirrors))
+    if not mirror_filter:
+        demisto.debug("Generic Message received, ignoring")
+        return
+
+    for mirror in mirror_filter:
+        if mirror['mirror_direction'] == 'FromDemisto' or mirror['mirror_type'] == 'none':
+            return
+
+        if not mirror['mirrored']:
+            demisto.info("Not mirrored yet")
+            # In case the investigation is not mirrored yet
+            mirror = mirrors.pop(mirrors.index(mirror))
+            if mirror['mirror_to'] and mirror['mirror_direction'] and mirror['mirror_type']:
+                investigation_id = mirror['investigation_id']
+                mirror_type = mirror['mirror_type']
+                auto_close = mirror['auto_close']
+                direction = mirror['mirror_direction']
+                if isinstance(auto_close, str):
+                    auto_close = bool(strtobool(auto_close))
+                demisto.info(f'Mirroring: {investigation_id}')
+                demisto.mirrorInvestigation(investigation_id,
+                                            f'{mirror_type}:{direction}', auto_close)
+                mirror['mirrored'] = True
+                mirrors.append(mirror)
+                set_to_integration_context_with_retries({'mirrors': mirrors},
+                                                        OBJECTS_TO_KEYS, SYNC_CONTEXT)
+
+        investigation_id = mirror['investigation_id']
+        await handle_text(ASYNC_CLIENT, investigation_id, text, user)  # type: ignore
+
+
+def fetch_context(force_refresh: bool = False) -> dict:
+    """
+    Fetches the integration instance context from the server if the CACHE_EXPIRY is smaller than the current epoch time
+    In the event that the cache is not expired, we return a cached copy of the context which has been stored in memory.
+    We can force the retrieval of the updated context by setting the force_refresh flag to True.
+    :param force_refresh: bool: Indicates if the context should be refreshed regardless of the expiry time.
+    :return: dict: Either a cached copy of the integration context, or the context itself.
+    """
+    demisto.info("Fetching context using fetch_context")
+    global CACHED_INTEGRATION_CONTEXT, CACHE_EXPIRY
+    if (CACHE_EXPIRY <= int(datetime.now(timezone.utc).timestamp() * 1000)) or force_refresh:
+        demisto.debug("Cached context has expired. Fetching new context")
```

Comment:
this is not correct if you passed "force_refresh"
```suggestion
        demisto.debug(f'Cached context has expired or forced refresh. forced refresh value is {force_refresh}. Fetching new context')
```